src/gpuarray_buffer_cuda.c

```diff
@@ -147,13 +148,49 @@ static int setup_lib(void) {
         return error_set(global_err, GA_IMPL_ERROR, "cuDriverGetVersion failed");
       major = tmp / 1000;
       minor = (tmp / 10) % 10;
+      #if defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64) || defined(__APPLE__)
+      /* We will dynamically search the right CUDA version only on Windows and Macintosh systems,
+      and only if user has not explicitely specified GPUARRAY_CUDA_VERSION. */
+      search_version = 1;
+      #endif
     } else {
       major = ver[0] - '0';
       minor = ver[1] - '0';
     }
+    /* NB: next line will cause problems if a CUDA 10.0 (or 9.11) is released in the future. */
```

Comment:
If we can look for all supported versions, then we can probably drop the environment variable, which would solve the problem.